lib/src/working_copy.rs

```diff
@@ -867,8 +867,16 @@ impl TreeState {
 
             // TODO: Check that the file has not changed before overwriting/removing it.
             match diff {
+                Diff::Removed(TreeValue::GitSubmodule(_old_id)) => {
+                    println!("warning: git submodule not removed at {:?}", path);
+                    self.file_states.remove(&path);
+                    stats.removed_files += 1;
+                }
                 Diff::Removed(_before) => {
-                    fs::remove_file(&disk_path).ok();
+                    fs::remove_file(&disk_path).map_err(|err| CheckoutError::IoError {
+                        message: format!("Failed to remove file {}", disk_path.display()),
+                        err,
+                    })?;
```

Comment:
I think we should continue checking out the rest of the files even if some fail. We could print a warning here for now.